index.bs

```diff
@@ -650,12 +634,14 @@ agent MUST run the following steps:
     :: |datagrams|
     : [=[[Session]]=]
     :: null
-1. Let |pullAlgorithm| be an action that runs [=pullDatagrams=] with |transport|.
-1. Let |writeAlgorithm| be an action that runs [=writeDatagrams=] with |transport|.
 1. [=ReadableStream/Set up=] |incomingDatagrams| with [=ReadableStream/set up/pullAlgorithm=]
-   set to |pullAlgorithm|, and [=ReadableStream/set up/highWaterMark=] set to 0.
+   set to an action that runs [=pullDatagrams=] with |transport|, and
+   [=ReadableStream/set up/highWaterMark=] set to 0.
 1. [=WritableStream/Set up=] |outgoingDatagrams| with [=WritableStream/set up/writeAlgorithm=]
-   set to |writeAlgorithm|.
+   set to an action that runs [=writeDatagrams=] with |transport|.
+1. [=ReadableStream/Set up=] |transport|'s [=[[IncomingBidirectionalStreams]]=] with
+   [=ReadableStream/set up/pullAlgorithm=] set to an action that runs [=pullBidirectionalStream=]
+   with |transport|, and [=ReadableStream/set up/highWaterMark=] set to 0.
```

Comment:
I think I prefer the before-version with explicit temporaries over inlining, and this is an example why: In the old prose it was clear that `highWatermark` is the second argument to the _set up_:

 1. "[Set up](https://streams.spec.whatwg.org/#readablestream-set-up) incomingDatagrams with [pullAlgorithm](https://streams.spec.whatwg.org/#readablestream-set-up-pullalgorithm) set to pullAlgorithm, and highWaterMark set to 0."

...not the second argument to the pull algorithm.

 1. "[Set up](https://streams.spec.whatwg.org/#readablestream-set-up) incomingDatagrams with [pullAlgorithm](https://streams.spec.whatwg.org/#readablestream-set-up-pullalgorithm) set to an action that runs pullDatagrams with transport, and highWaterMark set to 0."

The streams spec and [fetch](https://fetch.spec.whatwg.org/#concept-http-network-fetch) also seem to follow this pedantic pattern, so I think we should to..

Reply:
We set up multiple readable streams in this algorithm; datagram readable, and incoming bidirectional streams. I'll add incoming unidirectional streams soon. Hence we need three pullAlgorithms, and _pullAlgorithmForIncomingBidirectinalStream_ is way too long, which is why I inlined the creation of pullAlgorithm.

Reply:
OK, as both of you like naming them, I did that. By the way, which do you prefer, pullBidirectionalStream[Algorithm] or pullBidirectionalStreams[Algorithm]?

Reply:
I have a slight preference for pullBidirectionalStream, because only one BidirectionalStream is enqueued for each call to the algorithm.

Reply:
Thanks, then I may rename (pull|receive|write|send)Datagrams[Algorithm] in the future.